libminifi/include/utils/SmallString.h

```diff
@@ -0,0 +1,86 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <array>
+#include <ostream>
+#include <string>
+#include <utility>
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+template<size_t N>
+class SmallString : public std::array<char, N + 1> {
+ public:
+  operator std::string() const {  // NOLINT
```

Comment:
I would make this `explicit`, or maybe even create a separate conversion function, e.g. a `to_string` free function mimicking `std::to_string` with ADL. I'm not 100% sure that the example is a good idea, but it seems to be useful.

The important part is to avoid implicit conversion to `std::string`.

Reply:
in this case I would keep it implicit since using `SmallString` is purely an optimization and it should be kept as transparent as possible

Reply:
on a practical note, there are quite a number of places passing `::getUUIDStr()` where a `std::string` is expected, IMO we should rewrite all these functions to expect an `Identifier`

Reply:
ok